rfc/rfc006-distributed-registry.md

```diff
@@ -0,0 +1,344 @@
+* **name**: MUST contain the organization's commonly known name as string. Only alphanumeric and space characters are allowed,
+  it SHALL NOT be empty or only contain spaces. It MUST conform to the following regex: `[a-zA-Z0-9 ]+`.
+* **prfs**: (proofs) MUST contain an array with cryptographic proofs asserting the organization as being an actual care organization.
+  at the same it serves as commitment from the care organization that it uses the vendor's (identified by **vid**) software.
+  See section 6 for supported proof types.
+
+Example:
+```json
+{
+    "id": "1df8a26c-77df-47d0-862c-c35cd18af86f",
+    "vid": "urn:oid:1.3.6.1.4.1.54851.4:1234",
+    "name": "HomeSweetHome Thuiszorg",
+    "prfs": ["organisation proof"]
+}
+```
+
+#### Document type
+The document type when indicating an organization MUST be `registry/organization`.
+
+#### Validation
+When processing an organization registration or update it MUST be validated as follows:
+
+1. Assert all required fields are present.
+2. Assert all fields value formats are valid.
+3. For updates: assert only modifiable fields are updated.
+4. Assert the document is signed by the vendor identified by **vid**.
+5. Validate **prfs**:
+   * Assert there is at least one proof.
+   * Assert proofs are valid at the time they're introduced.
+   * Assert all proofs authenticate the same organisation.
+
+### 4.3. Service
+This document registers a service provided by an organization and SHOULD describe the endpoints that SHOULD be used for a specified Bolt (use case).
+
+#### Fields
+The following fields MUST be present:
+* **id**: MUST be a unique string identifying this endpoint.
+* **oid**: MUST contain the ID of the organization this endpoint belongs to.
+* **name**: MUST contain the name of the service as string.
+
+Example:
+```json
+{
+    "id": "5bcdc80b-ff63-4563-b919-d34fc38846c8",
+    "oid": "urn:oid:2.16.840.1.113883.2.4.6.1:1234",
+    "name": "eoverdracht"
+}
+```
+
+#### Document type
+The document type for services is `registry/service`.
+
+#### Validation
+When processing a organization registration or update it MUST be validated as follows:
+
+1. Assert all required fields are present.
+2. Assert all fields value formats are valid.
+3. For updates: assert only modifiable fields are updated.
+
+### 4.4. Endpoint
+This document registers an endpoint for an organization which CAN be used to exchange data with that organization.
+
+#### Fields
+The following fields MUST be present:
+* **id**: MUST be a unique string identifying this endpoint.
+* **oid**: MUST contain the ID of the organization this endpoint belongs to.
+* **type**: MUST be a URN describing the type of this endpoint in the form of `urn:oid:1.3.6.1.4.1.54851.2:type`.
+* **loc**: (location) MUST be a URL pointing to where the endpoint is exposed (e.g. `https://nuts.nl/some/path`)
+* **nbf**: (not before) MUST contain an RFC 3339 timestamp indicating from what date/time the endpoint SHOULD be considered valid.
```

Comment:
I don't know if this is needed when referencing from a service. A service will need it though..... A delete action is therefore also needed for endpoint/service removal

Reply:
This is actually not needed, maybe in the logical layer, not in the lower layers. It'll be orphaned, it may be deleted from a relational model. If it would be referenced in the future, it can be via document reference.